backend/data/src/main/kotlin/io/tolgee/model/Project.kt

```diff
@@ -97,6 +97,10 @@ class Project(
   @OneToMany(fetch = FetchType.LAZY, orphanRemoval = true, mappedBy = "project")
   var namespaces: MutableList<Namespace> = mutableListOf()
 
+  @OneToOne(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST])
+  @ActivityLoggedProp
+  var baseNamespace: Namespace? = null
```

Comment:
Could we name it `defaultNamespace`?

We also need to solve deleting - currently there is a logic that when user stops using the namespace it is automatically deleted. Now when the user selects the namespace as default it won't get deleted (which is correct I guess), but then it should get deleted once the user selects a different default (which is not happening now).

Reply:
> We also need to solve deleting - currently there is a logic that when user stops using the namespace it is automatically deleted. Now when the user selects the namespace as default it won't get deleted (which is correct I guess), but then it should get deleted once the user selects a different default (which is not happening now).

opps yes you are right. I haven't covered the last case. I'll cover that.